flink-runtime/src/test/java/org/apache/flink/runtime/checkpoint/CheckpointOptionsTest.java

```diff
@@ -55,9 +56,17 @@ public void testSavepoint() throws Exception {
         final byte[] locationBytes = new byte[rnd.nextInt(41) + 1];
         rnd.nextBytes(locationBytes);
 
+        final SnapshotType[] snapshotTypes = {
+            CHECKPOINT,
+            FULL_CHECKPOINT,
+            SavepointType.savepoint(SavepointFormatType.CANONICAL),
+            SavepointType.suspend(SavepointFormatType.CANONICAL),
+            SavepointType.terminate(SavepointFormatType.CANONICAL)
+        };
```

Comment:
Since it is not enum anymore there is highly likely to forget about this place when a new snapshot type will be added. Is it possible to avoid such a problem?

Reply:
Hmm... good point. So far I could not figure out a solution. However I am wondering how important it is we cover all different configurations. In the end we are pretty much testing that java serialization works well here. Is that crucial that we test all possible values?

Reply:
I would say it will be nice to test everything but of course, it is not so crucial. Perhaps, we should test the serialization of SavepointType separately using the constructor so if somebody adds a new parameter they will be forced to change the test.